src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,6 +183,21 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
                 var projects = await this.host.SonarQubeService.GetAllProjectsAsync(connection.Organization?.Key,
                     cancellationToken);
 
+                // The SonarQube client will limit the number of returned projects to 10K (hard limit on SonarQube side)
+                // but will no longer fail when trying to retrieve more. In the case where the project we want to bind to
+                // is not in the list AND the binding was already done (the .slconfig was manually created) then we are
+                // going to pretend we managed to retrieve this project from the list.
+                if (string.IsNullOrEmpty(connection.Password.ToUnsecureString()) &&
+                    projects.Count == 10000 &&
+                    !projects.Any(p => p.Key == this.host.VisualStateManager.BoundProjectKey))
+                {
+                    projects = new List<SonarQubeProject>
+                    {
+                        new SonarQubeProject(this.host.VisualStateManager.BoundProjectKey,
+                            this.host.VisualStateManager.BoundProjectName ?? this.host.VisualStateManager.BoundProjectKey)
+                    };
```

Comment:
We could just add the required project to the returned list rather than replacing the list of returned items completely.  Not sure which approach is preferable.

Reply:
I think it's better to add! At least we have a consistent UI when the user clicks the show all projects button.